src/core/sync/event.d

```diff
@@ -162,24 +164,42 @@ nothrow @nogc:
         }
     }
 
+    /// Set event to "signaled" if event is initialized
+    void setIfInitialized()
+    {
+        version (Windows)
+        {
+            if (m_event is null)
+                return;
+        }
+        else version (Posix)
+        {
+            if (!m_initalized)
+                return;
+        }
+        else
+            static assert(false);
+
+        set();
+    }
 
     /// Set the event to "signaled", so that waiting clients are resumed
     void set()
     {
         version (Windows)
         {
-            if (m_event)
-                SetEvent(m_event);
+            assert(m_event !is null);
```

Comment:
We have way (like version directive) to check if druntime precompiled or not?
If so, this would allow us to make a appropriate condition here and gradually all users would fix their programs.